squiggy/routes.py

```diff
@@ -99,32 +98,15 @@ def after_request(response):
             else:
                 app.logger.debug(log_message)
 
-        _create_cookies_as_needed(response)
         return response
 
 
 def _user_loader(user_id=None):
     from squiggy.lib.login_session import LoginSession
-    return LoginSession(user_id)
-
 
-def _create_cookies_as_needed(response):
-    if current_user.is_authenticated:
-        canvas_api_domain = current_user.course.canvas_api_domain
-        key = f'{canvas_api_domain}_{current_user.course.id}'
-        if not request.cookies.get(key):
-            response.set_cookie(
-                key=key,
-                value=str(current_user.user_id),
-                samesite='None',
-                secure=True,
-            )
-        key = f'{canvas_api_domain}_supports_custom_messaging'
-        if not request.cookies.get(key):
-            canvas = Canvas.find_by_domain(canvas_api_domain)
-            response.set_cookie(
-                key=key,
-                value=str(canvas.supports_custom_messaging),
-                samesite='None',
-                secure=True,
-            )
+    if not user_id:
+        course_site_uuid = request.headers.get('Squiggy-Course-Site-UUID')
+        if course_site_uuid:
+            cookie_value = request.cookies.get(course_site_uuid)
+            user_id = to_int(cookie_value) if cookie_value else None
+    return LoginSession(user_id)
```

Comment:
The use of 'Squiggy-Course-Site-UUID'  is inspired by the old SuiteC strategy of https://github.com/ets-berkeley-edu/suitec/blob/master/lib/col-lti/lib/rest.js#L72